lib/pdk/tests/unit.rb

```diff
@@ -6,20 +6,89 @@ module PDK
   module Test
     class Unit
       def self.cmd(_tests)
-        # TODO: actually run the tests
-        # cmd = 'rake spec'
-        # cmd += " #{tests}" if tests
-        cmd = 'pwd'
-        cmd
+        # TODO: test selection
+        [File.join(PDK::Util.module_root, 'bin', 'rake'), 'spec']
       end
 
-      def self.invoke(tests, report = nil)
+      def self.invoke(report, options = {})
         PDK::Util::Bundler.ensure_bundle!
+        PDK::Util::Bundler.ensure_binstubs!('rake')
 
-        puts _('Running unit tests: %{tests}') % { tests: tests }
+        tests = options.fetch(:tests)
 
-        output = PDK::CLI::Exec.execute(cmd(tests))
-        report.write(output) if report
+        cmd_argv = cmd(tests)
+        cmd_argv.unshift('ruby') if Gem.win_platform?
+
+        command = PDK::CLI::Exec::Command.new(*cmd_argv).tap do |c|
+          c.context = :module
+          c.add_spinner('Running unit tests')
+          c.environment['CI_SPEC_OPTIONS'] = '--format j'
+        end
+
+        PDK.logger.debug(_('Running %{cmd}') % { cmd: command.argv.join(' ') })
+
+        result = command.execute!
+
+        # TODO: cleanup rspec and/or beaker output
+        # Iterate through possible JSON documents until we find one that is valid.
+        json_result = nil
+
+        result[:stdout].scan(%r{\{(?:[^{}]|(?:\g<0>))*\}}x) do |str|
+          begin
+            json_result = JSON.parse(str)
+            break
+          rescue JSON::ParserError
+            next
+          end
+        end
+
+        raise PDK::CLI::FatalError, _('Unit test output did not contain a valid JSON result: %{output}') % { output: result[:stdout] } unless json_result
+
+        parse_output(report, json_result)
+
+        result[:exit_code]
+      end
+
+      def self.parse_output(report, json_data)
+        if json_data['messages'] && !json_data['messages'].empty?
+          json_data['messages'].each do |msg|
+            report.add_event(
+              source: 'rspec',
+              state: :error,
```

Comment:
I was looking at the rspec json output yesterday, and I was unsure if a message from rspec is necessarily an error; a quick browse of the rspec-core code suggests they're pretty neutral on what a _message_ is.
I think it's a fair assumption that a message from rspec is something we can consider an error.